src/main/java/ru/skypro/homework/controller/AdsController.java

```diff
@@ -0,0 +1,77 @@
+package ru.skypro.homework.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.enums.ParameterIn;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import jakarta.validation.Valid;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import ru.skypro.homework.dto.Ad;
+import ru.skypro.homework.dto.Ads;
+import ru.skypro.homework.dto.CreateOrUpdateAd;
+import ru.skypro.homework.dto.ExtendedAd;
+import ru.skypro.homework.service.impl.AdsServiceImpl;
+
+@RestController
+@RequestMapping("/ads")
+public class AdsController {
+    private final AdsServiceImpl adsService;
```

Comment:
```suggestion
    private final AdsService adsService;
```
инъекцию надо делать по интерфейсу, а не реализации